tests/Microsoft.Identity.Test.Integration.netfx/Infrastructure/HttpProxyManager.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Test.LabInfrastructure;
+
+namespace Microsoft.Identity.Test.Integration.NetFx.Infrastructure
+{
+    /// <remarks>
+    /// We invoke this class from different threads and they all use the same HttpClient.
+    /// To prevent race conditions, make sure you do not get / set anything on HttpClient itself,
+    /// instead rely on HttpRequest objects which are thread specific.
+    ///
+    /// In particular, do not change any properties on HttpClient such as BaseAddress, buffer sizes and Timeout. You should
+    /// also not access DefaultRequestHeaders because the getters are not thread safe (use HttpRequestMessage.Headers instead).
+    /// </remarks>
+    internal class ProxyHttpManager : IHttpManager
```

Comment:
Maybe this could extend HttpManager, to reduce code duplication, and the ExecuteAsync that has logic can be something like protected override.

Reply:
Yes we can, but the idea is to extend the proxyHttpManager for other http methods as well. that's the reason of implementing iHttpManager.